### PR TITLE
allow describe of workflow under deleted but existing namespace

### PR DESCRIPTION
**Pull request by jmbarzee**
## What changed?
allowed namespace states for describing mutable state of workflow.

## Why?
In some cases, deleted namespaces may not have their resources cleaned up right away. In these cases, its still useful to allow a workflow under the deleted namespace to be described.

## How did you test it?
- Hardcode variable to delay namespace cleanup
- Create workspace
- Create worker/workflow
- Delete namespace
- Wait for cache to expire
- Describe workflow

## Potential risks
No foreseeable risks. Preexisting protections against reading unauth/unauthz workflows still exists.

## Documentation
Documentation remains accurate

## Is hotfix candidate?
No

### CODE REVIEW

File: common/rpc/interceptor/namespace_validator.go [hunk 1]
```diff
@@ -62,10 +62,11 @@ var (
...
 
 	allowedNamespaceStates = map[string][]enumspb.NamespaceState{
-		"/temporal.api.workflowservice.v1.WorkflowService/StartWorkflowExecution":           {enumspb.NAMESPACE_STATE_REGISTERED},
-		"/temporal.api.workflowservice.v1.WorkflowService/SignalWithStartWorkflowExecution": {enumspb.NAMESPACE_STATE_REGISTERED},
-		"/temporal.api.operatorservice.v1.OperatorService/DeleteNamespace":                  {enumspb.NAMESPACE_STATE_REGISTERED, enumspb.NAMESPACE_STATE_DEPRECATED, enumspb.NAMESPACE_STATE_DELETED},
-		"/temporal.api.nexusservice.v1.NexusService/DispatchNexusTask":                      {enumspb.NAMESPACE_STATE_REGISTERED},
+		api.WorkflowServicePrefix + "StartWorkflowExecution":           {enumspb.NAMESPACE_STATE_REGISTERED},
+		api.WorkflowServicePrefix + "SignalWithStartWorkflowExecution": {enumspb.NAMESPACE_STATE_REGISTERED},
+		api.OperatorServicePrefix + "DeleteNamespace":                  {enumspb.NAMESPACE_STATE_REGISTERED, enumspb.NAMESPACE_STATE_DEPRECATED, enumspb.NAMESPACE_STATE_DELETED},
+		api.AdminServicePrefix + "DescribeMutableState":                {enumspb.NAMESPACE_STATE_REGISTERED, enumspb.NAMESPACE_STATE_DEPRECATED, enumspb.NAMESPACE_STATE_DELETED},
```
Comment (stephanos) [inline common/rpc/interceptor/namespace_validator.go:68]: Would it be helpful to add the "Why?" here as a code comment?

**Review by stephanos — APPROVED** [review]
What was your rational wrt writing an automated test or not?

File: common/rpc/interceptor/namespace_validator.go (hunk 1 above)
Comment (jmbarzee) [reply common/rpc/interceptor/namespace_validator.go:68]: I thought about it while trying to parse how to fix the issue, but ultimately I noticed that the line makes more sense to me one it was written. I'd be curious to hear your take, but for me it was a sort of "Yea, the adminService should allow visibility for a workflow no matter its state. I'm surprised we didn't in the first place" 

Additionally, I think I allowing the code to be self descriptive in this case may be preferable to a comment that may grow out of sync later.

**Review by stephanos — APPROVED** [review]

File: common/rpc/interceptor/namespace_validator.go (hunk 1 above)
Comment (stephanos) [reply common/rpc/interceptor/namespace_validator.go:68]: > for me it was a sort of "Yea, the adminService should allow visibility for a workflow no matter its state. I'm surprised we didn't in the first place"

I didn't think it was 100% obvious. For example, when I delete my AWS account, I won't have access anymore, but they are very likely deleting the resources asynchronously in the background.

But at the same time, it's not surprising behavior either. So I'm fine with how it is (and we always have `git blame`).

**Pull request merged**